pkg/chaosdaemon/tasks/chaosonprocess.go

```diff
@@ -0,0 +1,252 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package tasks
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/ChaosErr"
+)
+
+var ErrCanNotAdd = errors.New("can not add")
+var ErrCanNotAssign = errors.New("can not assign")
+
+// NewChaosOnProcess init ChaosOnProcess with values can not be Assigned.
```

Comment:
what is the meaning of `assign`?

Reply:
Assign is a interface `AssignChaosOnProcess`. It means these values do not need to copy & cost a lot to copy.